stm32-modules/heater-shaker/firmware/heater_task/heater_hardware.c

```diff
@@ -284,6 +377,95 @@ void HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef* hadc) {
     }
 }
 
+static uint16_t period_count = 0;
+void HAL_TIM_OC_DelayElapsedCallback(TIM_HandleTypeDef *htim)
+{
+    hw_internal* internal = (hw_internal*)HEATER_HW_HANDLE->hardware_internal;
+    if ((htim->Instance == TIM4) && (htim->Channel != HAL_TIM_ACTIVE_CHANNEL_3)) {
+        if ((internal->heatpad_cs_status != IDLE) || (internal->heatpad_cs_status != ERROR_OPEN_CIRCUIT) || (internal->heatpad_cs_status != ERROR_SHORT_CIRCUIT) || (internal->heatpad_cs_status != ERROR_OVERCURRENT)) {
+            period_count++;
+            if ((internal->heatpad_cs_status == RUNNING) && (period_count > 1000)) {
+                internal->heatpad_cs_status = PREP_CHECK;
+                uint8_t dac_val = (uint8_t) (0x0F); //0x0F is roughly 0.2V
+                HAL_DAC_SetValue(&internal->dac1, HEATPAD_CS_DAC_CHANNEL, DAC_ALIGN_8B_R, dac_val);
+                HAL_COMP_Stop(&internal->comp4);
+                internal->comp4.Init.Output = COMP_OUTPUT_NONE;
+                if (HAL_OK != HAL_COMP_Init(&internal->comp4)) {
+                    init_error();
+                }
+                HAL_COMP_Start(&internal->comp4);
+            } else if ((internal->heatpad_cs_status == PREP_CHECK) && (peripherals_ready())) {
+                if (can_heatpad_cs_open_check()) {
+                    internal->heatpad_cs_status = OPEN_CHECK_STARTED;
+                } else if (can_heatpad_cs_short_check()) {
+                    internal->heatpad_cs_status = SHORT_CHECK_STARTED;
+                }
+            } else if ((htim->Channel == HAL_TIM_ACTIVE_CHANNEL_2) && (internal->heatpad_cs_status == OPEN_CHECK_STARTED)) {
```

Comment:
Can we use `#define`'s with more specific names for the timer channels for each check?